src/globals.css

```diff
@@ -87,11 +87,20 @@
     @apply absolute left-0 top-0 block h-30 w-3 bg-main content-[''];
   }
 
+  /* Auth Styles */
+  .auth-form-section {
+    @apply flex flex-grow flex-col justify-center gap-8;
+  }
+
```

Comment:
> 출처 : [Reusing Styles](https://tailwindcss.com/docs/reusing-styles)
> Tailwind encourages a utility-first workflow, where designs are implemented using only low-level utility classes. This is a powerful way to avoid premature abstraction and the pain points that come with it.

tailwindcss는 유틸리티를 먼저 사용하여 작업할 것을 권장하고 있습니다. 물론 `@apply`와 같은 기능을 프로젝트가 성장함에 따라 다양한 장소에서 동일한 디자인을 재현할 때 공통 유틸리티 조합을 반복하는 경우를 위해 사용하기도 하죠. 실제로 reusing styles공식 문서 내용에서도 하나의 방법으로 소개하고 있습니다. 하지만 `@apply`에 대해서는 다음과 같은 이야기도 하고 있어요.

> 출처: [Reusing Styles > Avoiding premature abstraction](https://tailwindcss.com/docs/reusing-styles#avoiding-premature-abstraction)
> Whatever you do, don’t use `@apply` just to make things look “cleaner”.  

`@apply`를 단순히 코드를 깔끔하게 보이기 위해 사용하지 마세요

> If you start using `@apply` for everything, you are basically just writing CSS again and throwing away all of the workflow and maintainability advantages Tailwind gives you, for example:

`@apply`는 기본적으로 CSS를 다시 작성하고, tailwindcss에서 제공하는 모든 워크플로 및 유지 관리 이점을 버리는 행동이라고도 작성되어 있어요. 

현재 `auth-form-section`이 반복되는 영역을 살펴보았을 때,, form 영역에 끼워넣을 때 사용하는 section 태그에 사용하고 계시는더라구요? 만약에 끼워 넣는 section을 위에서 말씀드린 form에 한정 한다면 한 곳에서 사용하니 별도로 `@apply`로 추출할 필요는 없어 보입니다. 만약에 section 부분을 개별로 가져가신다면 사용해도 괜찮아 보이지만, 개인적으로는 form 한 곳에 두고, 필요한 입력 요소들을 넣어주기만 하는게 어떨까 싶습니다.

Reply:
오오??? 전혀 몰랐어요. 인터넷에 공통 부분 css 처리하는 방법을 찾아보면 저런 방식을 많이 소개해주셔서 당연히 사용하는 게 무조건 좋은 건 줄 알았습니다. 이렇게 또 알아가게 되네요. 공통 스타일 추출을 위해 많은 고민을 해봐야 할 것 같아요. 좋은 정보 감사합니다👍👍👍